js/Thread.js

```diff
@@ -1,28 +1,30 @@
-import {
-  NativeModules,
-  DeviceEventEmitter,
-} from 'react-native';
+import { NativeModules, DeviceEventEmitter } from "react-native";
 
 const { ThreadManager } = NativeModules;
 
 export default class Thread {
   constructor(jsPath) {
-    if (!jsPath || !jsPath.endsWith('.js')) {
-      throw new Error('Invalid path for thread. Only js files are supported');
+    if (!jsPath || !jsPath.endsWith(".js")) {
+      throw new Error("Invalid path for thread. Only js files are supported");
     }
 
     this.id = ThreadManager.startThread(jsPath.replace(".js", ""))
-      .then(id => {
-        DeviceEventEmitter.addListener(`Thread${id}`, (message) => {
-          !!message && this.onmessage && this.onmessage(message);
+      .then((id) => {
+        DeviceEventEmitter.addListener(`Thread`, (msg) => {
+          if (!msg) return;
+          const { message, id: threadId } = JSON.parse(msg);
+          if (id === threadId && this.onmessage)
+            this.onmessage(JSON.stringify(message));
```

Comment:
Have to re-stringify the message to preserve the current api/usage. We can callback the message here directly if we decide that is better. I feel we should preserve the api for now, esp. to make it simpler to switch to other thread/worker solutions if we ever need to.